src/kpro_connection.erl

```diff
@@ -270,16 +270,32 @@ get_tcp_mod(_SslOpts = true)  -> ssl;
 get_tcp_mod(_SslOpts = [_|_]) -> ssl;
 get_tcp_mod(_)                -> gen_tcp.
 
-maybe_upgrade_to_ssl(Sock, _Mod = ssl, SslOpts0, Timeout) ->
+%% If SslOpts contains {verify_peer, true}, we insert
+%% {server_name_indication, Host}. This is necessary as of OTP 20, to
+%% ensure that peer verification is done against the correct host name
+%% (otherwise the IP will be used, which is almost certainly
+%% incorrect).
+insert_server_name_indication(SslOpts, Host) when is_list(SslOpts) ->
+  case proplists:get_value(verify, SslOpts) of
+    verify_peer ->
+      [{server_name_indication, Host}|SslOpts];
+    _ ->
+      SslOpts
+  end;
+insert_server_name_indication(SslOpts, _) ->
+  SslOpts.
```

Comment:
`SslOpts` is ensured to be a list at line 291 so it will never get to this line.